### DISCUSSION TITLE
Absolute position of image in window?

### DISCUSSION THREAD

**Discussion by mgalos999** [emilk/egui#2095] [Q&A] [answered]
I can't seem to figure it out, but how can I draw a image to an absolute position on a window?
In theory I'd like to be able to specify the x and y of an image as negative values and it would be cropped accordingly.
Essentially I'd be able to move the image around with the window acting as a viewport.

Is this possible in egui?
Thanks in advance

**Comment by willser** [comment, marked as answer]
```
let texture: &egui::TextureHandle = self.texture.get_or_insert_with(|| {
    // Load the texture only once.
    ui.ctx()
        .load_texture("my-image", egui::ColorImage::example())
});

// Show the image:
let vec2 = texture.size_vec2();
let pos2 = Pos2::new(110., 200.);
ui.put(
    Rect::from_center_size(pos2, vec2),
    egui::Image::new(texture, vec2),
);

```
![mmexport1664780994705.png](https://user-images.githubusercontent.com/64425242/193519708-1b49ed15-480f-485f-9da2-14b1ab5c9c45.png)


If you want to change position,you could update variate  pos2.

**Comment by emilk** [comment]
`ui.painter().add(Shape::image(…));`

**Comment by mgalos999** [comment]
Excellent thanks guys this is exactly what I was looking for!
